File: 김현욱/4주차/10815.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int binary_search(vector<int> &v, int target) {
	int left = 0, right = v.size() - 1, mid;
	while(left <= right) {
		mid = (left + right) / 2;
		if (v[mid] == target)
			return mid;
		else if (v[mid] > target)
			right = mid - 1;
		else if (v[mid] < target)
			left = mid + 1;
	}
	return -1;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    
	vector<int> v;
	int n, m, num;
	
	cin >> n;
	
	for(int i = 0; i < n; i++) {
		cin >> num;
		v.push_back(num);
	}
	
	sort(v.begin(), v.end());
	
	cin >> m;
	
	for(int i = 0; i < m; i++) {
		cin >> num;
		if(binary_search(v, num) == -1)
			cout << "0 ";
		else
			cout << "1 ";
	}
	
	return 0;
}

File: 김경준/4주차/10815.py
#이진탐색 10815
n = int(input())
a = list(map(int, input().split()))
a.sort()

def binary_search(num):
    l = 0
    r = n-1
    while l <= r :
        mid = (l+r)//2
        if a[mid] == num :
            return 1
        elif a[mid] > num :
            r = mid - 1
            # 반 줄여주기 1
        else:
            l = mid + 1
            # 반 줄여주기 2
    return 0

input()
for num in list(map(int, input().split())):
    print(binary_search(num), end = ' ')
    
    
#시간초과 뜸
#10815
input()
n=list(map(int,input().split())) #갖고있는 카드 숫자 입력받기
#print(n)
input()
m=list(map(int,input().split())) #카드 숫자 입력받기
for i in m:
    if i in n:
        print(1,end=" ")
    else:
        print(0, end=" ")
